### DISCUSSION TITLE
flet apk problem white screen error

### DISCUSSION THREAD

**Discussion by merttalha** [flet-dev/flet#4196] [General]
I create my Flet application, but then I see a white screen and nothing else is visible.

**Comment by ap4499** [comment]
What platform? 

Can you share a minimal reproducible example? 

What version of Flet are you running? 

Is this Flet Build, or Flet Run?

**Reply by merttalha** [reply]
`import flet as ft
import requests


# Firebase yapılandırma bilgileri
FIREBASE_CONFIG = {
    "apiKey": "",
    "authDomain": "",
    "projectId": "",
    "storageBucket": "",
    "messagingSenderId": "",
    "appId": "",
    "databaseURL": ""
}



# Firebase Realtime Database URL'sini yapılandırın
FIREBASE_URL = ""

class LogInPage(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/login-user",
            vertical_alignment="center",  # Bileşenlerin dikey hizalanması
            horizontal_alignment="center",  # Bileşenlerin yatay hizalanması
        )
        self.page = page
        self.email_field = ft.TextField(width=258, height=45, hint_text="Telefon numarası, kullanıcı adı veya e-posta", bgcolor="#FAFAFA")
        self.password_field = ft.TextField(width=258, height=45, hint_text="Şifre", bgcolor="#FAFAFA", password=True)
        
        self.img = ft.Image(
            src="https://res.cloudinary.com/drcnpopuj/image/upload/v1728571535/Temel%20Resim%20klas%C3%B6r%C3%BC/8n91YnfPq0s2_zvowe8.png",
            height=61,
        )
        self.controls = [
            ft.Container(
                content=ft.Row(
                    alignment="center",
                    controls=[
                        ft.Column(
                            alignment="center",
                            controls=[
                                ft.Row(
                                    alignment="center",
                                    controls=[
                                        self.img
                                    ],
                                    width=350
                                ),
                                ft.Column(height=15),
                                ft.Row(
                                    alignment="center",
                                    controls=[
                                        self.email_field,
                                    ],
                                    width=350
                                ),
                                ft.Row(
                                    alignment="center",
                                    controls=[
                                        self.password_field, 
                                    ],
                                    width=350
                                ),
                                ft.Row(
                                    alignment="center",
                                    controls=[
                                        ft.Container(
                                            content=ft.Column(
                                                alignment="center",
                                                controls=[
                                                    ft.Row(
                                                        alignment="center",
                                                        controls=[
                                                            ft.Text(value="Giriş yap", color="white")
                                                        ]
                                                    )
                                                ],
                                            ),
                                            height=36,
                                            width=258,
                                            bgcolor="#0095f6",
                                            border_radius=4,
                                            on_click=self.login_user
                                        ),
                                    ],
                                    width=350
                                ),
                                ft.Row(
                                    controls=[
                                        ft.Column(width=35),
                                        ft.Text(value="Şifreni mi unuttun?")
                                    ],
                                    width=350
                                ),
                                
                            ]
                        )
                    ]
                ),
                height=390,
                width=350,
                border_radius=4,
                border=ft.border.all(2, ft.colors.BLACK),
            ),
            ft.Column(height=10),
            ft.Container(
                content=ft.Row(
                    alignment="center",
                    controls=[
                        ft.Column(
                            alignment="center",
                            controls=[
                                ft.Row(
                                    controls=[
                                        ft.Text(value="Hesabın yok mu?"),
                                        ft.Text(value="Kaydol", color="#0095f6")
                                    ]
                                )
                            ]
                        )
                    ]
                ),
                height=63,
                width=350,
                border_radius=4,
                border=ft.border.all(2, ft.colors.BLACK),
                on_click=lambda e: self.page.go("/create-user")
            ),
        ]

    def login_user(self, e):
        email = self.email_field.value
        password = self.password_field.value
        
        try:
            # Firebase Realtime Database'e e-posta ile sorgu yapma
            email_key = email.replace(".", ",")
            response = requests.get(f"{FIREBASE_URL}/users/{email_key}.json")
            
            if response.status_code == 200:
                user_data = response.json()

                if user_data and user_data['password'] == password:
                    self.page.client_storage.set("email", email)
                    self.page.client_storage.set("currentuser", email.replace("@gmail.com", ""))
                    self.page.client_storage.set("log", True)
                    self.page.client_storage.set("image", user_data['image'])
                    self.page.client_storage.set("name", user_data['name'])
                    self.page.client_storage.set("nickname", user_data['nickname'])
                    self.page.client_storage.set("url", user_data['url'])
                    self.page.client_storage.set("post", user_data['post'])
                    self.page.client_storage.set("followers", user_data['followers'])
                    self.page.client_storage.set("followed", user_data['followed'])  # Fixed key typo
                    self.page.client_storage.set("hem", 1)

                    print("Giriş başarılı!")  # Successful login message
                    self.page.go("/home")
                else:
                    print("E-posta veya şifre hatalı.")  # Invalid credentials message
            else:
                print("Kullanıcı verisi alınamadı.")  # Data fetch error message
        except Exception as e:
            print("Giriş yapılamadı:", e)  # Login failed message


# Firebase'e kullanıcı verilerini kaydetmek için kullanılan fonksiyon
def save_user_data(email, name, nickname, password, image, url, post, followers, followed, bio):
    data = {
        "name": name,
        "nickname": nickname,
        "password": password,
        "image": image,
        "url": url,
        "post": post,
        "followers": followers,
        "followed": followed,
        "bio": bio
    }

    # Firebase için geçerli bir anahtar oluşturmak için e-posta adresindeki noktaları virgülle değiştiriyoruz
    safe_email = email.replace(".", ",")
    
    # Veriyi Firebase'e kaydetmek için POST isteği
    response = requests.put(f"{FIREBASE_URL}users/{safe_email}.json", json=data)
    
    if response.status_code == 200:
        print("Kullanıcı başarıyla kaydedildi.")
    else:
        print("Kullanıcı kaydedilemedi:", response.text)



class CreatePage(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/create-user",
            vertical_alignment="center",
            horizontal_alignment="center",
            scroll=True
        )
        self.page = page
        self.email = ft.TextField(width=258, height=45, hint_text="Telefon numarası veya e-posta", bgcolor="#FAFAFA")
        self.name = ft.TextField(width=258, height=45, hint_text="Tam Adınız", bgcolor="#FAFAFA")
        self.password = ft.TextField(width=258, height=45, hint_text="Şifre", bgcolor="#FAFAFA")
        
        self.img = ft.Image(
            src="https://res.cloudinary.com/drcnpopuj/image/upload/v1728571535/Temel%20Resim%20klas%C3%B6r%C3%BC/8n91YnfPq0s2_zvowe8.png",
            height=61,
        )
        self.controls = [
            ft.Column(
                controls=[
                    ft.Container(
                        content=ft.Row(
                            alignment="center",
                            controls=[
                                ft.Column(
                                    alignment="center",
                                    controls=[
                                        ft.Row(
                                            alignment="center",
                                            controls=[
                                                
                                                self.img
                                            ],
                                            width=350
                                        ),
                                        ft.Column(height=10),
                                        ft.Column(
                                            controls=[
                                                ft.Row(
                                                    alignment="center",
                                                    controls=[
                                                        self.email,
                                                    ],
                                                    width=350
                                                ),
                                                ft.Row(
                                                    alignment="center",
                                                    controls=[
                                                        self.name,
                                                    ],
                                                    width=350
                                                ),
                                                ft.Row(
                                                    alignment="center",
                                                    controls=[
                                                        self.password,
                                                    ],
                                                    width=350
                                                ),
                                                  
                                            ]
                                        ),
                                        ft.Row(
                                            alignment="center",
                                            controls=[
                                                ft.Text(value="Arkadaşlarınızın fotoğraflarını ve videolarını görmek için kaydolun.", color="#737373", width=258, text_align="center"),
                                            ],
                                            width=350
                                        ),
                                        ft.Row(
                                            alignment="center",
                                            controls=[
                                                ft.Text(value="Hizmetimizi kullanan kişiler iletişim bilgilerinizi Divan'a yüklemiş olabilir. Daha fazla bilgi edin", color="#737373", width=258, text_align="center"),
                                            ],
                                            width=350
                                        ),
                                        ft.Row(
                                            alignment="center",
                                            controls=[
                                                ft.Text(value="Kaydolarak Şartlarımızı kabul etmiş olursunuz. Verilerinizi nasıl topladığımızı, kullandığımızı ve paylaştığımızı Gizlilik Politikamızdan ve çerezleri ve benzer teknolojileri nasıl kullandığımızı Çerez Politikamızdan öğrenin.", color="#737373", width=258, text_align="center"),
                                            ],
                                            width=350
                                        ),
                                        ft.Row(
                                            alignment="center",
                                            controls=[
                                                ft.Container(
                                                    content=ft.Column(
                                                        alignment="center",
                                                        controls=[
                                                            ft.Row(
                                                                alignment="center",
                                                                controls=[
                                                                    ft.Text(value="Kayıt ol", color="white")
                                                                ]
                                                            )
                                                        ],
                                                    ),
                                                    height=36,
                                                    width=258,
                                                    bgcolor="#0095f6",
                                                    border_radius=4,
                                                    on_click=self.register_user
                                                ),                                                
                                            ],
                                            width=350
                                        ),
                                    ]
                                )
                            ]
                        ),
                        height=655,
                        width=350,
                        border_radius=4,
                        border=ft.border.all(2, ft.colors.BLACK),
                    ),
                    ft.Column(height=10),
                    ft.Container(
                        content=ft.Row(
                            alignment="center",
                            controls=[
                                ft.Column(
                                    alignment="center",
                                    controls=[
                                        ft.Row(
                                            controls=[
                                                ft.Text(value="Zaten hesabın var mı?"),
                                                ft.Text(value="Giriş yap", color="#0095f6")
                                            ]
                                        )
                                    ]
                                )
                            ]
                        ),
                        height=63,
                        width=350,
                        border_radius=4,
                        border=ft.border.all(2, ft.colors.BLACK),
                        on_click=lambda e: self.page.go("/login-user")
                    ),             
                ],
                scroll="auto"  # This enables scrolling for the content
            ),
        ]


    def register_user(self, e):
        email = self.email.value
        image = "https://res.cloudinary.com/drcnpopuj/image/upload/v1727172601/profile_picture_amxta2.png"
        url = ""
        post = 0
        followers = 0 
        followed = 0 
        bio = "Ben " + str(self.name.value)

        # Alanların doldurulduğunu kontrol et
        if not email:
            print("Lütfen e-posta adresinizi girin.")
            return
        if not self.name.value:
            print("Lütfen adınızı girin.")
            return
        if not self.password.value:
            print("Lütfen şifrenizi girin.")
            return

        # E-posta adresinden takma ad (nickname) oluşturma
        if "@gmail.com" in email or "@hotmail.com" in email:
            self.page.client_storage.set("email", email)
            nickname = email.split("@")[0]  # @'den önceki kısmı takma ad olarak alıyoruz
        else:
            nickname = email  # Diğer alanlar için e-posta takma ad olarak kullanılacak

        name = self.name.value
        password = self.password.value

        try:
            # Kullanıcı zaten kayıtlı mı kontrol et (GET isteği)
            safe_email = email.replace(".", ",")
            existing_user_response = requests.get(f"{FIREBASE_URL}users/{safe_email}.json")
            
            if existing_user_response.status_code == 200 and existing_user_response.json() is not None:
                print("Kullanıcı zaten kayıtlı!")
                return
            
            # Eğer kullanıcı mevcut değilse yeni kullanıcı verilerini kaydet
            save_user_data(email, name, nickname, password, image, url, post, followers, followed, bio)
            
            print("Kullanıcı oluşturuldu ve kaydedildi!")
            self.page.go("/login-user")
        except Exception as e:
            print("Kullanıcı oluşturulamadı:", e)  # Kullanıcı oluşturulamadı mesajı




class searchuser(ft.Column):
    def __init__(self, username: str, profile_src: str, nickname: str, current_user: str):
        self.username = username
        self.current_user = current_user
        super().__init__(
            alignment="center",
            controls=[
                ft.Row(
                    controls=[
                        ft.Column(
                            controls=[
                                ft.Container(
                                    content=ft.Image(src=profile_src, height=44, width=44, border_radius=360),
                                ),
                            ],
                            width=50
                        ),
                        ft.Column(
                            controls=[
                                ft.Text(value="", weight="bold", spans=[ft.TextSpan(text=username, on_click=lambda e: self.load_page(nickname))]),
                                ft.Text(value="", spans=[ft.TextSpan(text=nickname, on_click=lambda e: self.load_page(nickname))]),
                            ],
                            spacing=0
                        ),
                    ]
                ),
            ],
            width=370
        )

    def load_page(self, nickname: str):
        self.page.client_storage.set("profileuser", nickname)
        self.page.client_storage.set("pagelook", True)
        self.page.go("/you-profile")

def get_usernames_from_emails(users_snapshot):
    """Extracts usernames, profile images, and nicknames from Firebase REST API response"""
    usernames = []
    for user_email, user_data in users_snapshot.items():
        email = user_email.replace(",", ".")  # Replace any commas with dots to form a proper email address
        username = user_data.get("name", "")
        profile_src = user_data.get("image", "")
        nickname = user_data.get("nickname", "")
        usernames.append((email, username, profile_src, nickname))
    return usernames



class Explore(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/explore",
            vertical_alignment="start",  
            horizontal_alignment="center",  
        )

        self.page = page
        self.search = ft.TextField(width=400, height=45, hint_text="Ara...", bgcolor="white")
        self.search_input_section = ft.Container(
            visible=True,
            content=ft.Row(
                controls=[
                    self.search,
                    ft.IconButton(icon=ft.icons.SEARCH, icon_size=35, icon_color="black", hover_color="white", highlight_color="white", on_click=lambda e: self.sync_users_data())  # Arama düğmesine basıldığında veri güncelle
                ]
            )
        )
        self.users_list = ft.Column()  # Kullanıcıları listelemek için alan

        # Sayfanın başlangıç durumu
        self.controls = [
            ft.Row(
                controls=[
                    ft.NavigationRail(
                        selected_index=None,
                        label_type=ft.NavigationRailLabelType.ALL,
                        min_width=100,
                        min_extended_width=400,
                        leading=ft.Container(
                            content=ft.Column(
                                alignment="center",
                                controls=[
                                    ft.Row(
                                        alignment="center",
                                        controls=[
                                            ft.Icon(name=ft.icons.ADD, color=ft.colors.BLUE, size=20)
                                        ]
                                    )
                                ]
                            ),
                            height=55,
                            width=55,
                            bgcolor="black",
                            border_radius=6,
                            border=ft.border.all(2, ft.colors.BLUE),
                            on_click=lambda e: self.page.go("/share-post")
                        ),
                        group_alignment=-0.9,
                        destinations=[
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.HOME_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label="Ana Sayfa",
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.EXPLORE_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label="Keşfet",
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.CHAT_BUBBLE_OUTLINE, color="BLUE"),
                                selected_icon_content=None,
                                label_content=ft.Text("Sohbetler"),
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.PERSON_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label_content=ft.Text("Profilim"),
                            ),
                        ],
                        on_change=lambda e: self.on_navigation_change(e),
                    ),
                    ft.VerticalDivider(width=1),
                    ft.Column(
                        scroll="auto",
                        controls=[
                            ft.Row(
                                controls=[ft.Column(
                                    controls=[self.search_input_section, self.users_list],
                                    height=300000000
                                )],
                                alignment="center"
                            ),
                        ], 
                        alignment=ft.MainAxisAlignment.START, expand=True
                    ),
                ],
                expand=True,
            )
        ]
    
    def on_navigation_change(self, e):
        if e.control.selected_index == 0:
            self.page.go("/home")
        if e.control.selected_index == 1:
            self.page.go("/explore")
        if e.control.selected_index == 2:
            self.page.go("/chats")
        if e.control.selected_index == 3:
            self.page.go("/notifications")
        if e.control.selected_index == 3:
            self.page.go("/my-profile")

    def update_search_results(self):
        """Arama sonucunu güncelle"""
        search_text = self.search.value.lower()
        filtered_users = [user for user in self.usernames if search_text in user[0].lower()]
        self.display_users(filtered_users)

    def sync_users_data(self):
        """Firebase'den kullanıcıları al ve UI'yi güncelle (REST API kullanarak)"""
        db_url = f"{FIREBASE_CONFIG['databaseURL']}users.json"  # Firebase REST API URL
        response = requests.get(db_url)

        if response.status_code == 200:
            users_snapshot = response.json()  # Kullanıcıları JSON formatında al
            self.usernames = get_usernames_from_emails(users_snapshot)  # Kullanıcı adlarını al
            self.update_search_results()  # Arama sonucunu güncelle ve göster
        else:
            print("Kullanıcıları alırken hata oluştu:", response.text)

    def display_users(self, usernames):
        """Kullanıcıları ekranda göster"""
        self.users_list.controls.clear()  # Eski kullanıcıları temizle
        current_user = self.page.client_storage.get("current_user")  # Geçerli kullanıcıyı al
        for email, username, profile_src, nickname in usernames:
            self.users_list.controls.append(searchuser(username=username, profile_src=profile_src, nickname=nickname, current_user=current_user))
        self.page.update()  # Ekranı güncelle

class Comments(ft.Column):
    def __init__(self, username: str, profile_src: str, comment: str):
        super().__init__(
            alignment="center",
            controls=[
                ft.Row(
                    controls=[
                        ft.Column(
                            controls=[
                                ft.Container(
                                    content=ft.Image(src=profile_src, height=44, width=100, border_radius=360),
                                    padding=ft.Padding(left=10, right=10),  # 10 birim boşluk
                                ),
                            ],
                            width=50
                        ),
                        ft.Column(
                            controls=[
                                ft.Text(value=username, weight="bold"),
                                ft.Text(value=comment),
                            ],
                        ),
                    ]
                ),
            ],
            width=370
        )



class Post(ft.Column):
    def __init__(self, post_id: str, image_src: str, username: str, profile_src: str, current_user: str):
        self.post_id = post_id
        self.username = username
        self.profile_src = profile_src
        self.current_user = current_user

        # Like buttons
        self.like_button = ft.IconButton(
            icon=ft.icons.FAVORITE_BORDER, icon_size=35, icon_color="black", hover_color="white", highlight_color="white",
            visible=True, on_click=lambda e: self.like_post(post_id)
        )
        self.liked_button = ft.IconButton(
            icon=ft.icons.FAVORITE, icon_size=35, icon_color="black", hover_color="white", highlight_color="white",
            visible=False, on_click=lambda e: self.unlike_post(post_id)
        )

        # Comment buttons
        self.comment_button = ft.IconButton(
            icon=ft.icons.MODE_COMMENT_OUTLINED, icon_size=35, icon_color="black", hover_color="white", highlight_color="white",
            visible=True, on_click=self.toggle_comment_section
        )
        self.comment_filled_button = ft.IconButton(
            icon=ft.icons.MODE_COMMENT, icon_size=35, icon_color="black", hover_color="white", highlight_color="white",
            visible=False, on_click=self.toggle_comment_section
        )

        # Comment input area
        self.comment = ft.TextField(width=347, height=45, hint_text="Yorum...", bgcolor="#FAFAFA")
        self.comment_input_section = ft.Container(
            visible=False,
            content=ft.Row(
                controls=[
                    self.comment,
                    ft.IconButton(icon=ft.icons.SEND_OUTLINED, icon_size=35, icon_color="black", hover_color="white", highlight_color="white", on_click=lambda e: self.add_comment())
                ]
            )
        )

        # Comment display section
        self.comments_container = ft.Container(
            visible=False,
            content=ft.Column(
                controls=[
                    ft.Text(value="Yorumlar", size=18, weight="Bold"),
                ],
                width=390,
                scroll=True,
                height=400
            ),
            width=404,
            border_radius=4,
            border=ft.border.all(2, ft.colors.BLACK),
            padding=10
        )

        super().__init__(
            alignment="center",
            controls=[
                ft.Row(controls=[ft.Column(height=10)]),
                ft.Container(
                    content=ft.Row(
                        controls=[
                            ft.Column(width=0),
                            ft.Container(
                                content=ft.Row(
                                    alignment="center",
                                    controls=[
                                        ft.Column(
                                            alignment="center",
                                            controls=[ft.Image(src=profile_src, height=40, width=40, border_radius=360)],
                                        )
                                    ],
                                ),
                                border_radius=360,
                                border=ft.border.all(2, ft.colors.BLUE),
                                height=45,
                                width=45
                            ),
                            ft.Text(size=18, spans=[ft.TextSpan(text=username, on_click=lambda e: self.page.client_storage.set("userdata", username))]),
                        ],
                    ),
                    width=404,
                    border_radius=4,
                    border=ft.border.all(2, ft.colors.BLACK),
                    height=60
                ),
                ft.Container(
                    content=ft.Row(
                        controls=[ft.Image(src=image_src, height=400, width=400)],
                        spacing=0
                    ),
                    width=404,
                    border_radius=4,
                    border=ft.border.all(2, ft.colors.BLACK),
                ),
                ft.Container(
                    content=ft.Row(
                        controls=[
                            ft.Row(
                                controls=[
                                    self.like_button,
                                    self.liked_button,
                                    self.comment_button,
                                    self.comment_filled_button,
                                ]
                            ),
                        ],
                        spacing=0
                    ),
                    width=404,
                    border_radius=4,
                    border=ft.border.all(2, ft.colors.BLACK),
                ),
                self.comment_input_section,
                self.comments_container,
            ]
        )

        # Check the like status
        self.check_like_status()

    def toggle_comment_section(self, e):
        """Toggle the visibility of the comment section and comment input field."""
        self.comment_input_section.visible = not self.comment_input_section.visible
        self.comments_container.visible = not self.comments_container.visible

        # Switch between the comment buttons
        self.comment_button.visible = not self.comment_button.visible
        self.comment_filled_button.visible = not self.comment_filled_button.visible

        if self.comments_container.visible:
            self.load_comments()

        self.update()

    def load_comments(self):
        """Fetch comments from the REST API and display them."""
        try:
            response = requests.get(f"{FIREBASE_URL}comments/{self.post_id}.json")
            if response.status_code == 200:
                comments_data = response.json()
                self.display_comments(comments_data)
            else:
                print(f"Yorumlar alınırken hata oluştu: {response.status_code}")
        except Exception as e:
            print(f"Yorumları yüklerken bir hata oluştu: {str(e)}")

    def display_comments(self, comments_data):
        """Display fetched comments in the UI."""
        self.comments_container.content.controls.clear()

        if comments_data:
            for comment_id, comment_data in comments_data.items():
                self.comments_container.content.controls.append(
                    ft.Row(
                        controls=[
                            ft.Image(src=comment_data.get("profile_src", "default_profile_image_url"), height=40, width=40, border_radius=360),
                            ft.Text(f"{comment_data.get('username', 'Bilinmeyen Kullanıcı')}: {comment_data.get('comment', 'Yorum bulunmuyor.')}")
                        ]
                    )
                )
        else:
            self.comments_container.content.controls.append(
                ft.Text(value="Henüz yorum yapılmamış.", size=16, color=ft.colors.GREY)
            )

        self.update()

    def add_comment(self):
        """Add a new comment using the REST API."""
        comment_text = self.comment.value
        if comment_text:
            comment_data = {
                "username": self.current_user,
                "profile_src": self.page.client_storage.get("image"),
                "comment": comment_text
            }
            try:
                response = requests.post(f"{FIREBASE_URL}comments/{self.post_id}.json", json=comment_data)
                if response.status_code == 200:
                    print("Yorum başarıyla eklendi.")
                    self.load_comments()  # Reload comments after adding a new one
                    self.comment.value = ""
                    self.update()
                else:
                    print(f"Yorum eklenirken hata oluştu: {response.status_code}")
            except Exception as e:
                print(f"Yorum eklenirken bir hata oluştu: {str(e)}")

    def check_like_status(self):
        """Check if the post is liked by the current user via the REST API."""
        try:
            response = requests.get(f"{FIREBASE_URL}likes/{self.post_id}/{self.current_user}.json")
            if response.status_code == 200 and response.json():
                self.update_like_buttons(True)
            else:
                self.update_like_buttons(False)
        except Exception as e:
            print(f"Beğenme durumu kontrol edilirken hata oluştu: {str(e)}")

    def update_like_buttons(self, is_liked: bool):
        """Update the visibility of the like and liked buttons based on the like status."""
        self.like_button.visible = not is_liked
        self.liked_button.visible = is_liked
        self.update()

    def like_post(self, post_id: str):
        """Like the post using the REST API."""
        like_data = {
            "username": self.current_user,
            "profile_src": self.page.client_storage.get("image"),
        }
        try:
            response = requests.put(f"{FIREBASE_URL}likes/{post_id}/{self.current_user}.json", json=like_data)
            if response.status_code == 200:
                self.update_like_buttons(True)
            else:
                print(f"Beğenme işlemi başarısız: {response.status_code}")
        except Exception as e:
            print(f"Beğenme işlemi sırasında hata oluştu: {str(e)}")

    def unlike_post(self, post_id: str):
        """Unlike the post using the REST API."""
        try:
            response = requests.delete(f"{FIREBASE_URL}likes/{post_id}/{self.current_user}.json")
            if response.status_code == 200:
                self.update_like_buttons(False)
            else:
                print(f"Beğenmeyi geri alma başarısız: {response.status_code}")
        except Exception as e:
            print(f"Beğenmeyi geri alırken hata oluştu: {str(e)}")





class HomePage(ft.View):
    def __init__(self, page: ft.Page, current_user: str):
        super().__init__(
            route="/home",
            vertical_alignment="",
            horizontal_alignment="center",
        )
        self.page = page
        self.image = ""
        self.current_user = current_user
        self.posts = []  # List to hold posts
        self.story = [story(username="ekle", profile_src=self.image)]
        
        # Load user data
        self.load_user()

        # Fetch and display posts
        self.load_posts()

        # Layout
        self.controls = [
            ft.Row(
                controls=[
                    ft.NavigationRail(
                        selected_index=None,
                        label_type=ft.NavigationRailLabelType.ALL,
                        min_width=100,
                        min_extended_width=400,
                        leading=ft.Container(
                            content=ft.Column(
                                alignment="center",
                                controls=[
                                    ft.Row(
                                        alignment="center",
                                        controls=[
                                            ft.Icon(name=ft.icons.ADD, color=ft.colors.BLUE, size=20)
                                        ]
                                    )
                                ]
                            ),
                            height=55,
                            width=55,
                            bgcolor="black",
                            border_radius=6,
                            border=ft.border.all(2, ft.colors.BLUE),
                            on_click=lambda e: self.page.go("/share-post")
                        ),
                        group_alignment=-0.9,
                        destinations=[
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.HOME_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label="Ana Sayfa",
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.EXPLORE_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label="Keşfet",
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.CHAT_BUBBLE_OUTLINE, color="BLUE"),
                                selected_icon_content=None,
                                label_content=ft.Text("Sohbetler"),
                            ),
                            ft.NavigationRailDestination(
                                icon_content=ft.Icon(ft.icons.PERSON_OUTLINED, color="BLUE"),
                                selected_icon_content=None,
                                label_content=ft.Text("Profilim"),
                            ),
                        ],
                        on_change=lambda e: self.on_navigation_change(e),
                    ),
                    ft.VerticalDivider(width=1),
                    ft.Column(
                        scroll="auto",
                        controls=[
                            ft.Column(
                                controls=[
                                    ft.Column(height=0),
                                    ft.Container(
                                        content=ft.Row(
                                            controls=[
                                                ft.Column(width=4),
                                                ft.Text(value="Ana Sayfa", size=20, color="BLUE"),
                                                ft.Column(width=10)
                                            ]
                                        ),
                                        bgcolor="black",
                                        height=55,
                                        border_radius=8,
                                        border=ft.border.all(2, ft.colors.BLUE)
                                    )
                                ],
                                height=60
                            ),
                            ft.Row(
                                scroll="auto",
                                controls=[
                                    # Storyler burada görünecek
                                ]
                            ),
                            ft.Row(
                                alignment="center",
                                controls=[
                                    ft.Column(
                                        scroll="auto",
                                        controls=[
                                            ft.Column(
                                                scroll="auto",
                                                controls=self.posts,  # Postlar burada görünecek
                                                height=3000000000
                                            ),
                                        ]
                                    )                                    
                                ]
                            ),
                        ], 
                        alignment=ft.MainAxisAlignment.START, expand=True
                    ),
                ],
                expand=True,
            )
        ]

    def on_navigation_change(self, e):
        if e.control.selected_index == 0:
            self.page.go("/home")
        if e.control.selected_index == 1:
            self.page.go("/explore")
        if e.control.selected_index == 2:
            self.page.go("/chats")
        if e.control.selected_index == 3:
            self.page.go("/my-profile")

    def load_posts(self):
        """Fetch posts from the database and display them."""
        try:
            response = requests.get(f"{FIREBASE_URL}posts.json")
            if response.status_code == 200:
                posts_data = response.json()
                if posts_data:
                    self.posts = []
                    for post_id, post_data in posts_data.items():
                        if isinstance(post_data, dict) and "image_src" in post_data and "username" in post_data and "profile_src" in post_data:
                            new_post = Post(
                                post_id=post_id,
                                image_src=post_data["image_src"],
                                username=post_data["username"],
                                profile_src=post_data["profile_src"],
                                current_user=self.current_user
                            )
                            self.posts.append(new_post)
                    self.update_posts()
            else:
                print(f"Error loading posts: {response.status_code}")
        except Exception as e:
            print(f"Error loading posts: {str(e)}")

    def update_posts(self):
        """Update the UI with the fetched posts."""
        self.controls[0].controls[2].controls[2].controls[0].controls = self.posts
        self.page.update()

    def load_user(self):
        """Load user data from the database."""
        email = self.page.client_storage.get("email")
        try:
            email_key = email.replace(".", ",")
            user_data = requests.get(f"{FIREBASE_URL}users/{email_key}.json").json()
            self.page.client_storage.set("image", user_data['image'])
            self.image = user_data['image']
            self.page.client_storage.set("currentuser", email.replace("@gmail.com", ""))
        except Exception as e:
            print(f"Error loading user data: {e}")





class story(ft.Column):
    def __init__(self, username: str, profile_src: str):
        super().__init__(
            alignment="center",
            controls=[
                ft.Row(
                    alignment="center",
                    controls=[
                        ft.Column(
                            alignment="center",
                            spacing=0,
                            controls=[
                                ft.Container(
                                    content=ft.Row(
                                        alignment="center",
                                        controls=[
                                            ft.Column(
                                                alignment="center",
                                                controls=[
                                                    ft.Image(src=profile_src, height=60, width=60, border_radius=360)
                                                ],
                                                spacing=0
                                            )
                                        ],
                                        spacing=0
                                    ),
                                    border_radius=360,
                                    border=ft.border.all(2, ft.colors.BLUE),
                                ),
                                ft.Text(value=username, color="black", text_align="center", width=80)
                            ]
                        )
                    ]
                )
            ]
        )















class Error(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/reload",
            vertical_alignment="center",
            horizontal_alignment="center",
        )
        self.page = page


        # Controls for the profile editing page
        self.controls = [
            ft.Row(
                controls=[
                    ft.Row(
                        controls=[
                            ft.Column(
                                controls=[
                                    ft.Row(
                                        controls=[
                                            ft.Column(expand=True),
                                            ft.Icon(name=ft.icons.HIGHLIGHT_OFF_SHARP,size=100,color="#0095f6"),
                                            ft.Column(expand=True),
                                        ],
                                        width=350
                                    ),
                                    
                                    ft.Column(height=10),
                                    ft.Text("İstemci depolamasının daha stabil çalışması için",color="#0095f6",width=350,text_align="center"),
                                    ft.Text("tarayıcınızdaki sekmeyi kapatıp tarayıcınızı",color="#0095f6",width=350,text_align="center"),
                                    ft.Text("birkaç kez açıp kapatın.",color="#0095f6",width=350,text_align="center"),
                                ]
                            )
                        ],
                        alignment="center"
                    )                    
                ],
                alignment="center"
            ),

        ]



def main(page: ft.Page):
    page.bgcolor = "#FFFFFF"
    page.theme_mode = "light"
    current_user = page.client_storage.get("currentuser")
    
    create = CreatePage(page)
    login = LogInPage(page)
    reload = Error(page)
    home = HomePage(page, current_user)
    explore = Explore(page)

    
    # Router function
    def route_change(route):
        # Clear current views
        page.views.clear()
        if page.route == "/create-user":
            page.views.append(create)
        if page.route == "/login-user":
            page.views.append(login)
        if page.route == "/reload":
            page.views.append(reload)
        if page.route == "/home" or page.client_storage.get("log") == True:
            page.views.append(home)
        if page.route == "/explore":
            page.views.append(explore)


        page.update()

    # Trigger the route change on route change
    page.on_route_change = route_change

    # Start with the intended route
    page.go("/login-user")  # Start directly with SharePost if needed



ft.app(target=main,assets_dir="assets")
`

![image](https://github.com/user-attachments/assets/79c325cd-94b3-4eb4-8b14-781422e4c567)

Even though the platform I use the APK on is Android, I encounter a black screen. I print the APK from github.
